### ISSUE TITLE
Model::afterFetch() is not fired on Model::refresh() 

### ISSUE THREAD

**Issue by Gamblt** [phalcon/cphalcon#12220]
### Expected and Actual Behavior

When i refresh() a model, afterfetch is not fired.
If refresh() function "re-querying the record from the database"  it's necessary to prepare new data using afterfetch().

> Provide minimal script to reproduce the issue

I have Users table with afterFetch() function

``` php
public function afterFetch()
{
    $empty = empty($this->first_name) && empty($this->last_name);
    $this->full_name = ($empty) ? $this->nick_name : $this->first_name.' '.$this->last_name;
}
```

So. When i update first_name or last_name save it and refresh() model, the full name is still the same.
It's works good if i use 

``` php
$user = Users::findFirst($user->id) 
```

instead of 

``` php
$user->refresh();
```
- Phalcon version: 2.0.11
- PHP Version: 5.6.25-2
- Operating System: Ubuntu 14.04

**Comment by Jurigag** [comment]
Well this is not an issue i think. This is just how refresh works. Why fire `afterFetch` when you have already fetched model ? You just want to refresh it, for sure it shouldn't be in framework. If you want `afterFetch` fired then just extend refresh method.

**Comment by michanismus** [comment]
Hmm, I think it needs to be fired. Otherwise the event should be named "afterFirstFetch"...

**Comment by sergeyklay** [comment]
@Gamblt  `The 2.0.x` branch is not longer supported. Could you please try `3.0.x` version?

**Comment by aat2703** [comment]
Any news on this issue?

**Comment by sergeyklay** [comment]
Implemented in the 3.4.x branch. Thank you for the report
